modules/tensor_mechanics/src/actions/CohesiveZoneActionBase.C

```diff
@@ -43,6 +43,9 @@ CohesiveZoneActionBase::validParams()
   params.addParam<MooseEnum>("strain", strainType, "Strain formulation");
 
   // Advanced
+  params.addParam<bool>("use_automatic_differentiation",
+                        false,
+                        "Whether to use automatic differentiation to compute the Jacobian");
```

Comment:
Does it work if CZM is AD but TM is not AD? If not, maybe we should grab it from TM action or throw a warning.

Reply:
All the interface materials/kernels should be AD/nonAD consistent.
Likewise, all the bulk materials/kernels should be AD/nonAD consistent.
So we can mix and match in that sense... and yes, it should work if CZM is AD but TM is nonAD.